Refactor project design, remove duplicate code, implement temporary package structure

ryanw-ny opened the pull request:
This removes any bloated or duplicate code from the application, it also moves things around to add more organization and make it easier to work with in the future. Temporary package structure has been implemented although it will probably be revised down the road as more features get implemented.

This shouldn't impact performance or how the program currently functions, this just cleans things up. All sub-activities have been migrated to their own package and can be easily found in ```mozilla.org.webmaker.activity``` excluding the default activity and web maker activities. 

This closes issue  #1501

/cc @thisandagain

thisandagain commented on app/src/main/java/mozilla/org/webmaker/WebmakerActivity.java:
```diff
@@ -34,10 +34,9 @@ protected void onCreate(Bundle savedInstanceState) {
-            view.destroy();
-            view = null;
-        }
+        assert view != null;
```
This assert is removed during runtime which makes this brittle, no?

thisandagain commented on app/src/main/java/mozilla/org/webmaker/WebmakerApplication.java:
```diff
@@ -3,12 +3,16 @@
 public class WebmakerApplication extends Application {
 
     // Singleton
-    private static WebmakerApplication singleton;
```
Good catch. :+1:

thisandagain commented on app/src/main/java/mozilla/org/webmaker/fragment/WebviewFragment.java:
```diff
@@ -68,34 +44,9 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
-            mWebView.destroy();
-            mWebView = null;
-        }
+        assert mWebView != null;
```
Same question here.

thisandagain commented:
Looks great. Just one small question / concern regarding the use of an assertion within the `onDestroyView` lifecycle event.

ryanw-ny commented:
@thisandagain Good catch with the assertions! I went ahead and replaced all of them with regular conditional statements like suggested and added a check to see if the ActionBar is null and if so throw a error for debugging purposes.

thisandagain commented on app/src/main/java/mozilla/org/webmaker/MainActivity.java:
```diff
@@ -29,7 +29,7 @@ protected void onCreate(Bundle savedInstanceState) {
         // Set up the action bar.
         final ActionBar actionBar = getActionBar();
-        assert actionBar != null;
+        if (actionBar == null) throw new NullPointerException("ActionBar has returned null!");
```
:+1:

thisandagain commented:
Awesome. Looks great. R+ :+1: :cat2: :cake: :boom:
